packages/sanity/src/_internal/cli/server/constants.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * This script takes the import map from the `#__imports` script tag,
+ * modifies relevant URLs that match the sanity-cdn hostname by replacing
+ * the existing timestamp in the sanity-cdn URLs with a new runtime timestamp,
+ * and injects the modified import map back into the HTML.
+ *
+ * This will be injected into the HTML of the user's bundle.
+ *
+ * Note that this is in a separate constants file to prevent "Cannot access
+ * before initialization" errors.
+ */
+export const TIMESTAMPED_IMPORTMAP_INJECTOR_SCRIPT = `<script>
+  // auto-generated script to add import map with timestamp
+  const { imports = {}, ...rest } = JSON.parse(document.getElementById('__imports').textContent);
```

Comment:
Can we gracefully handle a missing imports script? Edge case, but figure it's worth the few extra bytes

```suggestion
  const importsScript = document.getElementById('__imports');
  const { imports = {}, ...rest } = importsScript ? JSON.parse(importsScript.textContent) : {};
```